tests/net/conn_mgr_conn/src/main.c

```diff
@@ -0,0 +1,706 @@
+}
+
+/* Verify that flag get/set fail and behave as expected respectively for invalid ifaces and
+ * invalid flags.
+ */
+ZTEST(conn_mgr_conn, test_flags_invalid)
+{
+	int invalid_flag = CONN_MGR_NUM_IF_FLAGS;
+
+	/* Verify set failure for invalid ifaces / flags */
+	zassert_equal(conn_mgr_if_set_flag(ifnull, CONN_MGR_IF_PERSISTENT, true), -ENOTSUP,
+		"Setting persistence flag should fail for ifnull");
+	zassert_equal(conn_mgr_if_set_flag(ifnone, CONN_MGR_IF_PERSISTENT, true), -ENOTSUP,
+		"Setting persistence flag should fail for ifnone");
+	zassert_equal(conn_mgr_if_set_flag(ifa1, invalid_flag, true), -EINVAL,
+		"Setting invalid flag should fail for ifa1");
+
+	/* Verify get graceful behavior for invalid ifaces / flags */
+	zassert_false(conn_mgr_if_get_flag(ifnull, CONN_MGR_IF_PERSISTENT),
+		"Getting persistence flag should yield false for ifnull");
+	zassert_false(conn_mgr_if_get_flag(ifnone, CONN_MGR_IF_PERSISTENT),
+		"Getting persistence flag should yield false for ifnone");
+	zassert_false(conn_mgr_if_get_flag(ifa1, invalid_flag),
+		"Getting invalid flag should yield false for ifa1");
+}
+
+
+/* Verify that timeout get/set functions operate correctly (A/B) */
+ZTEST(conn_mgr_conn, test_timeout)
+{
+	struct conn_mgr_conn_binding *ifa1_binding = conn_mgr_if_get_binding(ifa1);
+
+	/* Try setting timeout */
+	zassert_equal(conn_mgr_if_set_timeout(ifa1, 99), 0,
+				"Setting timeout should succeed for ifa1");
+
+	/* Verify success */
+	zassert_equal(conn_mgr_if_get_timeout(ifa1), 99,
+				"Timeout should be set to 99 for ifa1");
+
+	/* Verify that the conn struct agrees, since this is what implementations may use */
+	zassert_equal(ifa1_binding->timeout, 99,
+				"Timeout set should affect conn struct");
+
+	/* Try unsetting timeout */
+	zassert_equal(conn_mgr_if_set_timeout(ifa1, CONN_MGR_IF_NO_TIMEOUT), 0,
+				"Unsetting timeout should succeed for ifa1");
+
+	/* Verify success */
+	zassert_equal(conn_mgr_if_get_timeout(ifa1), CONN_MGR_IF_NO_TIMEOUT,
+				"Timeout should be unset for ifa1");
+
+	/* Verify that the conn struct agrees, since this is what implementations may use */
+	zassert_equal(ifa1_binding->timeout, CONN_MGR_IF_NO_TIMEOUT,
+				"Timeout unset should affect conn struct");
+}
+
+/* Verify that timeout get/set fail and behave as expected respectively for invalid ifaces */
+ZTEST(conn_mgr_conn, test_timeout_invalid)
+{
+	/* Verify set failure */
+	zassert_equal(conn_mgr_if_set_timeout(ifnull, 99), -ENOTSUP,
+		"Setting timeout should fail for ifnull");
+	zassert_equal(conn_mgr_if_set_timeout(ifnone, 99), -ENOTSUP,
+		"Setting timeout should fail for ifnone");
+
+	/* Verify get graceful behavior */
+	zassert_equal(conn_mgr_if_get_timeout(ifnull), CONN_MGR_IF_NO_TIMEOUT,
+		"Getting timeout should yield CONN_MGR_IF_NO_TIMEOUT for ifnull");
+	zassert_equal(conn_mgr_if_get_timeout(ifnone), CONN_MGR_IF_NO_TIMEOUT,
+		"Getting timeout should yield CONN_MGR_IF_NO_TIMEOUT for ifnone");
+}
+
+ZTEST(conn_mgr_conn, test_temp2)
+{
+}
```

Comment:
Missed this earlier, should we remove this if there is no actual test implemented?

Reply:
@glarsennordic @rlubos

Reply:
Yes, this should be removed, thank you for catching it